pkg/controller/logstorage/secrets/secret_controller.go

```diff
@@ -276,23 +267,45 @@ func (r *SecretSubController) Reconcile(ctx context.Context, request reconcile.R
 	// Provision secrets and the trusted bundle into the cluster.
 	hdler := utils.NewComponentHandler(reqLogger, r.client, r.scheme, ls)
 
-	// Create Elasticsearch secrets.
-	esTrustedBundle := elasticKeys.trustedBundle(clusterCM)
-	if err = hdler.CreateOrUpdateOrDelete(ctx, elasticKeys.component(esTrustedBundle), r.status); err != nil {
-		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error creating / updating resource", err, reqLogger)
-		return reconcile.Result{}, err
-	}
+	// Determine if Kibana should be enabled for this cluster.
+	kibanaEnabled := !operatorv1.IsFIPSModeEnabled(install.FIPSMode) && !r.multiTenant
 
-	if kibanaEnabled {
-		// Render the key pair and trusted bundle into the Kibana namespace.
-		if err = hdler.CreateOrUpdateOrDelete(ctx, elasticKeys.kibanaComponent(esTrustedBundle), r.status); err != nil {
+	// Internal ES modes:
+	// - Zero-tenant: everything installed in tigera-elasticsearch/tigera-kibana Namespaces. We need a single trusted bundle in each.
+	// - Single-tenant: everything installed in tigera-elasticsearch/tigera-kibana Namespaces. We need a single trusted bundle in each.
+	//
+	// External ES modes:
+	// - Single-tenant: everything installed in tigera-elasticsearch Namespace. We need a single trusted bundle.
+	// - Multi-tenant: nothing installed in tigera-elasticsearch Namespace, we need a trusted bundle per tenant.
```

Comment:
This is an assumption that I think is true? For multi-tenant external ES, there is no need for a trusted bundle in the tigera-elasticsearch Namespace because we don't run any components there. Hopefully a reviewer can confirm or deny that assertion.

Reply:
@asincu @tmjd would like your confirmation on this one

Reply:
I would expect this to be true. Also, for external ES mode we always have mTLS enabled, while for internal ES mode we do not have it enabled. We need to load `tigera-secure-es-http-certs-public` inside the trusted bundle because it contains the CA for external elastic.

Reply:
> Also, for external ES mode we always have mTLS enabled, while for internal ES mode we do not have it enabled. We need to load tigera-secure-es-http-certs-public inside the trusted bundle because it contains the CA for external elastic.

yes, that is already handled in this code later on but I don't this is relevant to this particular comment.

You can see it here: https://github.com/tigera/operator/blob/5cdf4100c68d2dbac70affdfe7f87ab2611be58f/pkg/controller/logstorage/secrets/secret_controller.go#L498-L503

Reply:
I'm surprised you're asking me because I would have thought you would know better than I would.
I'm not aware of a current need for a bundle in multi-tenant mode.

Reply:
@tmjd I should! I'm doubting my own memory of things given the state of some of this code... so wanting to double and triple check.